algorithms/ml/linear_regression.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
```

Comment:
There is no other use of external libraries throughout the codebase. Functionaluty should be implemented using plain vanilla Python.